packages/cardpay-sdk/sdk/scheduled-payment-module.ts

```diff
@@ -918,4 +920,137 @@ export default class ScheduledPaymentModule {
+    if (!moduleAddress) throw new Error('moduleAddress must be provided');
+    if (!tokenAddress) throw new Error('tokenAddress must be provided ');
+    if (!amount) throw new Error('amount must be provided');
+    if (!payeeAddress) throw new Error('payeeAddress must be provided');
+    if (feeFixedUSD == undefined) throw new Error('feeFixedUSD must be provided');
+    if (feePercentage == undefined) throw new Error('feePercentage must be provided');
+    if (!executionGas) throw new Error('executionGas must be provided');
+    if (!maxGasPrice) throw new Error('maxGasPrice must be provided');
+    if (!gasTokenAddress) throw new Error('gasTokenAddress must be provided ');
+    if (!salt) throw new Error('salt must be provided');
+    if (payAt == null && recurringDayOfMonth == null && recurringUntil == null)
+      throw new Error('When payAt is null, recurringDayOfMonth and recurringUntil must have a value');
+
+    let { onTxnHash } = txnOptions ?? {};
+    let module = new this.web3.eth.Contract(ScheduledPaymentABI as AbiItem[], moduleAddress);
+    let executeScheduledPaymentData;
+    if (recurringUntil) {
+      executeScheduledPaymentData = module.methods[
+        'executeScheduledPayment(address,uint256,address,((uint256),(uint256)),uint256,uint256,address,string,uint256,uint256,uint256)'
+      ](
+        tokenAddress,
+        amount,
+        payeeAddress,
+        {
+          fixedUSD: {
+            value: FEE_BASE.mul(new BN(feeFixedUSD)).toString(),
+          },
+          percentage: {
+            value: FEE_BASE.mul(new BN(feePercentage)).toString(),
+          },
+        },
+        executionGas,
+        maxGasPrice,
+        gasTokenAddress,
+        salt,
+        recurringDayOfMonth,
+        recurringUntil,
+        gasPrice
+      ).encodeABI();
+    } else {
+      executeScheduledPaymentData = module.methods[
+        'executeScheduledPayment(address,uint256,address,((uint256),(uint256)),uint256,uint256,address,string,uint256,uint256)'
+      ](
+        tokenAddress,
+        amount,
+        payeeAddress,
+        {
+          fixedUSD: {
+            value: FEE_BASE.mul(new BN(feeFixedUSD)).toString(),
+          },
+          percentage: {
+            value: FEE_BASE.mul(new BN(feePercentage)).toString(),
+          },
+        },
+        executionGas,
+        maxGasPrice,
+        gasTokenAddress,
+        salt,
+        payAt,
+        gasPrice
+      ).encodeABI();
+    }
+
+    let executeScheduledPaymentTx = {
+      to: moduleAddress,
+      value: '0',
+      data: executeScheduledPaymentData,
+      operation: Operation.CALL,
+    };
+    let txnHash;
+    try {
+      txnHash = await sendTransaction(this.web3, executeScheduledPaymentTx);
+      if (typeof onTxnHash === 'function') {
+        await onTxnHash(txnHash);
+      }
+      return await waitUntilTransactionMined(this.web3, txnHash);
+    } catch (e: any) {
+      // UnknownHash: payment details generate unregistered spHash
```

Comment:
@jurgenwerk I document the function error here since the extractSendTransaction is a general function that can be used for non-scheduled payment module contracts.

Reply:
Perfect, thank you!